src/connect.rs

```diff
@@ -826,13 +841,17 @@ mod socks {
                 &password,
             )
             .await
+            .map_err(|_| crate::error::builder("failure connecting socks"))?
         } else {
-            let s = Socks5Stream::connect(socket_addr, (host.as_str(), port)).await;
-            TcpStream::from_std(s.into_inner(), &reactor::Handle::default())
-                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
+            Socks5Stream::connect(socket_addr, (host.as_str(), port))
+                .await
+                .map_err(|_| crate::error::builder("failure connecting socks"))?
         };
 
-        Ok((Box::new(s) as super::Conn, Connected::new()))
+        Ok(super::Conn {
+            inner: Box::new( stream.into_inner() ),
+            is_proxy: true,
```

Comment:
I believe `is_proxy` should be false. I realize it's not super clear, since we did just connect over a proxy, but it's meant to say "is plain text HTTP proxy". This tells hyper whether the URI should be written in origin-form (`GET /just/a/path HTTP/1.1`) or absolute-form (`GET http://foo.bar/and/a/path HTTP/1.1`).

Reply:
Ah okay, yeah, that was not really obvious to me. I'll make the changes tomorrow.

Reply:
I tried to add a doc comment to the `is_proxy` field to make that more explicit. The `pin_project!` macro around the `Conn` declaration can't handle doc comments on fields, though, it seems. I tried both `/// ...` and `#[doc="..."]` syntax.

I'm adding the doc comment to the struct instead, hope that is okay for now. I'll mark it for you to review.